src/Networking/NexusMods.Networking.Steam/ManifestParser.cs

```diff
@@ -0,0 +1,40 @@
+using NexusMods.Abstractions.Hashes;
+using NexusMods.Abstractions.Steam.DTOs;
+using NexusMods.Abstractions.Steam.Values;
+using NexusMods.Paths;
+using SteamKit2;
+
+namespace NexusMods.Networking.Steam;
+
+public static class ManifestParser
+{
+    public static Manifest Parse(DepotManifest manifest)
+    {
+        return new Manifest
+        {
+            ManifestId = ManifestId.From(manifest.ManifestGID),
+            TotalCompressedSize = Size.From(manifest.TotalCompressedSize),
+            TotalUncompressedSize = Size.From(manifest.TotalUncompressedSize),
+            CreationTime = manifest.CreationTime,
+            DepotId = DepotId.From(manifest.DepotID),
+            Files = manifest.Files!.Select(file => new Manifest.FileData
+                {
+                    Path = file.FileName,
+                    Size = Size.From(file.TotalSize),
+                    Hash = Sha1.From(file.FileHash),
+                    Chunks = file.Chunks.Select(chunk => new Manifest.Chunk
+                        {
+                            ChunkId = Sha1.From(chunk.ChunkID),
+                            Offset = chunk.Offset,
+                            CompressedSize = Size.From(chunk.CompressedLength),
+                            UncompressedSize = Size.From(chunk.UncompressedLength),
+                            Checksum = Crc32.From(chunk.Checksum),
+                        }
+                    ).ToArray(),
+                }
+            ).ToArray(),
+        };
```

Comment:
```suggestion
        var files = manifest.Files is null ? [] : manifest.Files.Select(file => new Manifest.FileData
        {
            Path = file.FileName,
            Size = Size.From(file.TotalSize),
            Hash = Sha1.From(file.FileHash),
            Chunks = file.Chunks.Select(chunk => new Manifest.Chunk
            {
                ChunkId = Sha1.From(chunk.ChunkID),
                Offset = chunk.Offset,
                CompressedSize = Size.From(chunk.CompressedLength),
                UncompressedSize = Size.From(chunk.UncompressedLength),
                Checksum = Crc32.From(chunk.Checksum),
            }).ToArray(),
        }).ToArray();

        return new Manifest
        {
            ManifestId = ManifestId.From(manifest.ManifestGID),
            TotalCompressedSize = Size.From(manifest.TotalCompressedSize),
            TotalUncompressedSize = Size.From(manifest.TotalUncompressedSize),
            CreationTime = manifest.CreationTime,
            DepotId = DepotId.From(manifest.DepotID),
            Files = files,
        };
```

Reply:
Why this edit? It's pretty much exactly the same code, except the original version matches the output structure and this is builds the files first then uses them later. Feels like a change for little impact.

Reply:
This just makes sure a null value in `manifest.Files`  doesn't blow up the system.